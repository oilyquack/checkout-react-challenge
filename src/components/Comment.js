import React from "react";

function Comment({ review }) {
  function starGen(rating) {
    let ratingHTML = "";
    for (let i = 1; i <= 5; i++) {
      if (i <= rating) {
        ratingHTML += `⭑`;
      } else {
        ratingHTML += `☆`;
      }
    }
    return ratingHTML;
  }

  return (
    <div className="app__comments__comment">
      <h2>{review.name}</h2>
      {starGen(review.rating)}
      <p>{review.comment}</p>
    </div>
  );
}

export default Comment;
