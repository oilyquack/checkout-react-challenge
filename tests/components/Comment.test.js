import React from "react";
import renderer from "react-test-renderer";
import Comment, { starGen } from "../../src/components/Comment";

it("matches the snapshot", () => {
  const review = {
    name: "Connan Mockasin",
    comment: "Amazing!",
    rating: 5
  };
  const tree = renderer.create(<Comment review={review} />);
  expect(tree).toMatchSnapshot();
});

test("star generation returns the right amount of stars", function() {
  const expected = "★★★☆☆";

  const result = starGen(3);

  expect(result).toBe(expected);
});
