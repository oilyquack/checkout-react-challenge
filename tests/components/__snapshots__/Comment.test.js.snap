// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="app__comments__comment"
>
  <h2>
    Connan Mockasin
  </h2>
  ★★★★★
  <p>
    Amazing!
  </p>
</div>
`;
